service/worker/failovermanager/workflow_test.go

```diff
@@ -635,5 +623,10 @@ func (s *failoverWorkflowTestSuite) prepareTestActivityEnv() (*testsuite.TestAct
 	s.activityEnv.SetWorkerOptions(worker.Options{
 		BackgroundActivityContext: context.WithValue(context.Background(), failoverManagerContextKey, ctx),
 	})
-	return s.activityEnv, mockResource, controller
+
+	s.T().Cleanup(func() {
+		mockResource.Finish(s.T())
+	})
```

Comment:
can we do this in the `resource.NewTest` func, so it's picked up everywhere like gomock, or does that cause too many issues at the moment?